pkg/repository/v1_repository.go

```diff
@@ -379,6 +380,23 @@ func (r *V1Repository) updateComponentManifest(id string) (*v1manifest.Component
 	return &component, nil
 }
 
+// DownloadComponent downloads a component with specific version from repository
+func (r *V1Repository) DownloadComponent(
+	component string,
+	version v0manifest.Version,
+	versionItem *v1manifest.VersionItem,
+) error {
+	cr, err := r.downloadComponent(versionItem)
+	if err != nil {
+		return err
+	}
+
+	resName := fmt.Sprintf("%s-%s", component, version)
+
+	filename := fmt.Sprintf("%s-%s-%s", resName, r.GOOS, r.GOARCH)
```

Comment:
OT: not related to this PR, but I think we should change the term of `GOOS` and `GOARCH` to more general ones, as these are not exactly the Go buildin variables.